src/cozmo_companion/assistant.py

```diff
@@ -103,47 +127,82 @@ def _listen(self):
                     keywords=KEYWORDS,
                     keywords_threshold=KEYWORDS_THRESHOLD,
                 ).get_result()
-
+                # Extract the transcribed text from the result
                 user_speech_text = speech_result["results"][0]["alternatives"][0][
                     "transcript"
                 ]
 
             return user_speech_text
+        # Handle exceptions from the IBM service
         except ApiException as ex:
             logging.error(
                 "Method failed with status code " + str(ex.code) + ": " + ex.message
             )
 
+    # @tool
+    def get_feedback_inquiry(self, user_text: str = "") -> Optional[str]:
```

Comment:
```suggestion
    def get_feedback_inquiry(self, user_text: str = "") -> None | str:
```

Reply:
I am using Optional[str] for Python 3.9 since I read the other way is for 3.10 and above. I plan to upgrade though, and then I can use the other type annotation.